### PR TITLE
fix: detach tensors before passing to LAMMPS in ML-IAP wrapper

### LINKED ISSUE #133: ML-IAP interface Allegro+lammps bug about "fix langevin" for controlling temperature locally
After trying the Allegro_0.7.1+lammps with ML-IAP interface, using the "fix langevin" to set different temperatures for different regions of a system for calculate the thermal conductivity. The local temperatures cannot be established and with the wrong f_fixID which is used to compute the accumulated input heat energy with time. The lammps can run but with physical non-sense reseut. Saying that there is strange bug becuase i have compared with previous non-ML-IAP versions (the AOTI version)  Allegro+lammps with  the same lammps script but just different pair_style and the previouse version functions correctly. The problem may be about the ML model or the lammps ML_IAP interface. Also found other ML-IAP interfaced ML+lammps (MACE_0.3.13+ML-IAP interfaced lammps) have the same problem. The problem may more likely be the lammps ML-IAP interface but still cannot conclusively comfirm which part of the process the problem comes form.

### PR DESCRIPTION
**Pull request by manishpaulish**
(maybe) fixes mir-group/allegro#133

All three tensors passed to LAMMPS at the end of `compute_forces` still carry `requires_grad=True` -`nequip_atomic_energies` and `nequip_total_energy` because they're downstream of `edge_vectors` which has `requires_grad_(True)` set on line 221, and `edge_forces` because it comes from `torch.autograd.grad`.

When thermostats like `fix langevin` add random and drag forces directly to the per-atom force array in-place after each force evaluation, the in-place modification can corrupt PyTorch's computational graph if those arrays are still attached to it. This explains the temperature explosion to 100K+ observed in the linked issue with `fix langevin` but not with `fix nvt`, since Nosé-Hoover modifies velocities rather than the force array directly.

The `UserWarning: Converting a tensor with requires_grad=True to a scalar` on line 178 is a symptom of the same issue.

This fix adds `.detach()` before handing tensors to LAMMPS. It has no effect on correctness since gradients are never needed past this point, and no effect on performance.

### CODE REVIEW

File: nequip/integrations/lammps_mliap/lmp_mliap_wrapper.py [hunk 1]
```diff
@@ -5,9 +5,9 @@
 import os
 from pathlib import Path
 
-from nequip.data import AtomicDataDict
+from nequip.data import AtomicDataDic
```
Comment (cw-tan) [inline nequip/integrations/lammps_mliap/lmp_mliap_wrapper.py]: This should not be changed

File: nequip/integrations/lammps_mliap/lmp_mliap_wrapper.py [hunk 2]
```diff
@@ -5,9 +5,9 @@
...
 from nequip.data.transforms.neighborlist import NeighborListPruneTransform
-from nequip.nn import graph_model
+from nequip.nn import graph_mode
```
Comment (cw-tan) [inline nequip/integrations/lammps_mliap/lmp_mliap_wrapper.py]: This should not be changed

**Review by cw-tan — DISMISSED** [review]

**Comment by cw-tan** [comment]
Hi @manishpaulish thanks for the PR, though I think maybe there were some unintended changes to the imports?

**Comment by manishpaulish** [comment]
Sorry about that. Those were unintended changes. Fixed in 337dcf5.

**Comment by cw-tan** [comment]
Thanks, merged!

**Comment by manishpaulish** [comment]
Thanks for the review and merge. I've been digging into the NequIP/Allegro codebase and would love to keep contributing. Are there any open issues or areas the group is actively working on where an extra pair of hands would be useful?

**Comment by cw-tan** [comment]
Hi @manishpaulish , I wouldn't say that there are active problems -- things have been relatively stable these days. I think the best kind of contribution is where you're pushing the infrastructure to its limits with your own project, found gaps and are happy to contribute some changes that could be useful to others. 

In terms of nice-to-haves, it would be convenient for us to have a DataModule for the datasets supported by https://github.com/jla-gardner/load-atoms so that we can easily access them, which entails implementing auto-download features and making sure the format conversion is correct for several datasets. That way more people can more easily access various datasets directly from the NequIP infrastructure. 

They also seem to have some open issues, so you might also find opportunities to contribute there.
